DeezyMatch/candidateRanker.py

```diff
@@ -275,36 +300,45 @@ def candidate_ranker(input_file_path="default", query_scenario=None, candidate_s
             query_candidate_pd['s2'] = all_candidates
             query_candidate_pd['s2_orig'] = all_candidates_orig
+
             query_candidate_pd['label'] = "False"
     
             # Compute cosine similarity
             cosine_sim = cosine_similarity(vecs_query[iq:(iq+1)].detach().cpu().numpy(), 
                                            vecs_candidates.detach().cpu().numpy()[orig_id_candis])
             cosine_dist = 1. - cosine_sim
     
-            if not pretrained_model_path in [False, None]:
-                all_preds = candidate_conf_calc(query_candidate_pd, 
-                                                model, 
-                                                train_vocab, 
-                                                dl_inputs, 
-                                                cutoffs=(id_1_neigh - id_0_neigh))
-                query_candidate_pd['dl_match'] = all_preds.detach().cpu().numpy()
-    
+            if use_predict == True:
+                if not pretrained_model_path in [False, None]:
+                    all_preds = candidate_conf_calc(query_candidate_pd, 
+                                                    model, 
+                                                    train_vocab, 
+                                                    dl_inputs, 
+                                                    cutoffs=(id_1_neigh - id_0_neigh))
+                    query_candidate_pd['dl_match'] = all_preds.detach().cpu().numpy()
+        
+                else:
+                    query_candidate_pd['dl_match'] = [None]*len(query_candidate_pd)
             else:
                 query_candidate_pd['dl_match'] = [None]*len(query_candidate_pd)
     
-    
             query_candidate_pd['faiss_dist'] = found_neighbours[0][0, id_0_neigh:id_1_neigh]
             query_candidate_pd['cosine_dist'] = cosine_dist[0] 
             query_candidate_pd['s1_orig_ids'] = orig_id_queries 
             query_candidate_pd['s2_orig_ids'] = orig_id_candis 
+
+            # Filter out candidates that have a larger string length difference than the one allowed:
+            if isinstance(length_diff, int):
+                query_candidate_pd = query_candidate_pd[abs(query_candidate_pd["s1"].str.len() - query_candidate_pd["s2"].str.len()) <= length_diff]
     
             if ranking_metric.lower() in ["faiss"]:
                 query_candidate_filtered_pd = query_candidate_pd[query_candidate_pd["faiss_dist"] <= selection_threshold]
             elif ranking_metric.lower() in ["cosine"]:
                 query_candidate_filtered_pd = query_candidate_pd[query_candidate_pd["cosine_dist"] <= selection_threshold]
             elif ranking_metric.lower() in ["conf"]:
-                if not pretrained_model_path in [False, None]:
+                if use_predict == False:
+                    sys.exit(f"ranking_metric: {ranking_metric} is selected, but use_predict is set to {use_predict}")
```

Comment:
Should we move this up? Before doing any computations?

Reply:
Good point, I moved it: https://github.com/Living-with-machines/DeezyMatch/pull/121/commits/daa3ee6d214bcabd081d0e3cedb42b7387a78955